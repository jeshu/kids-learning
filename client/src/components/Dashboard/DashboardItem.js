import React from 'react'
import { Card, CardActionArea, CardActions, Button, CardContent, Typography, CardMedia } from "@material-ui/core"


function DashboardItem({path, title, desc, onPreview, onLaunch, id}) {
    return (title ? 
        <Card data-testid="dashboard-item">
            <CardActionArea>
                <CardMedia
                    image={path}
                    title={title}
                />
                <CardContent>
                    <Typography gutterBottom variant="h5">{title}</Typography>
                    <Typography variant="body2" color="textSecondary" component="p">{desc}</Typography>
                </CardContent>
            </CardActionArea>
            <CardActions>
                <Button title="preview" onClick={()=>onPreview({id})}>
                    Preview
                </Button>
                <Button title="launch" onClick={()=>{onLaunch({id})}}>
                    Launch
                </Button>
            </CardActions>
        </Card> : "");
}

export default DashboardItem
