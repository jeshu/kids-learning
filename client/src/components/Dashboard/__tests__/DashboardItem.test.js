import React from "react";
import DashboardItem from "../DashboardItem";
import { render, fireEvent } from "@testing-library/react";

describe('tests for component DashboardItem', () => {
    const props = {
        id: "someId",
        path:"someImagePath",
        title:"someDummyTitle",
        desc: "some dummy description"
    }

    it('should render without errors but not buttons', () => {
        const { container, queryAllByRole } = render(<DashboardItem />);
        expect(queryAllByRole("button")).toHaveLength(0);
        expect(container).toMatchSnapshot();
    });

    it('should render without errors with buttons ', () => {
        const { container, queryAllByRole } = render(<DashboardItem {...props}/>);
        expect(queryAllByRole("button")).toHaveLength(3);
        expect(container).toMatchSnapshot();
    });
    
    it('should call the preview method from props with id', () => {
        props.onPreview = jest.fn();
        const {getByTitle} = render(<DashboardItem {...props}/>);
        fireEvent.click(getByTitle("preview"));
        expect(props.onPreview).toHaveBeenCalledTimes(1);
        expect(props.onPreview).toHaveBeenCalledWith({id:"someId"});
    });

    it('should call the onLaunch of props with id', () => {
        props.onLaunch = jest.fn();
        const {getByTitle} = render(<DashboardItem {...props}/>);
        fireEvent.click(getByTitle("launch"));
        expect(props.onLaunch).toHaveBeenCalledTimes(1);
        expect(props.onLaunch).toHaveBeenCalledWith({id:"someId"});
        
    });
    
});
