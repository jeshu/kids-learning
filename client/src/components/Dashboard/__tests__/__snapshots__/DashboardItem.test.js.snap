// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for component DashboardItem should render without errors but not buttons 1`] = `
<div>
  
</div>
`;

exports[`tests for component DashboardItem should render without errors with buttons  1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="dashboard-item"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardMedia-root makeStyles-cardImage-2"
        style="background-image: url(someImagePath);"
        title="someDummyTitle"
      />
      <div
        class="MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          someDummyTitle
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          some dummy description
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        title="preview"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Preview
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        title="launch"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Launch
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
